awx/api/views/__init__.py

```diff
@@ -209,6 +209,26 @@ def api_exception_handler(exc, context):
     return exception_handler(exc, context)
 
 
+class LabelList(DeleteLastUnattachLabelMixin, SubListCreateAttachDetachAPIView):
+    def post(self, request, *args, **kwargs):
+        # If a label already exists in the database, attach it instead of erroring out
+        # that it already exists
+        if not getattr(self, 'label_filter', None):
```

Comment:
If it's a reverse ForeignKey then `parent_key` is the back-reference. It seems that we don't have something similar for many-to-many relationship sublists.

But it seems that we could get it in an algorithmic way.

```
In [6]: Schedule._meta.get_field('labels').remote_field.name
Out[6]: 'schedule_labels'

In [7]: JobTemplate._meta.get_field('labels').remote_field.name
Out[7]: 'unifiedjobtemplate_labels'

In [8]: WorkflowJobNode._meta.get_field('labels').remote_field.name
Out[8]: 'workflowjobnode_labels'
```

It's not that I'd prefer this generally, but here, it would avoid the need for the `return Response(...)` part, which is not good practice and also fairly unreachable by testing.